spec/view_objects/result_filters/qualification_view_spec.rb

```diff
@@ -0,0 +1,100 @@
+require_relative "../../../app/view_objects/result_filters/qualification_view"
+
+module ResultFilters
+  RSpec.describe QualificationView do
+    describe "qts_only_checked?" do
+      subject { described_class.new(params: params).qts_only_checked? }
+
+      context "when QtsOnly param not present" do
+        let(:params) { { qualifications: "Other,PgdePgceWithQts" } }
+        it { is_expected.to eq(false) }
+      end
+
+      context "when QtsOnly param is present" do
+        let(:params) { { qualifications: "QtsOnly,PgdePgceWithQts" } }
+        it { is_expected.to eq(true) }
+      end
+
+      context "when qualifications is empty" do
+        let(:params) { { qualifications: "" } }
+        it { is_expected.to eq(true) }
+      end
+
+      context "when qualifications is nil" do
+        let(:params) { {} }
+        it { is_expected.to eq(true) }
+      end
+    end
+
+    describe "pgde_pgce_with_qts_checked" do
+      subject { described_class.new(params: params).pgde_pgce_with_qts_checked? }
+
+      context "when PgdePgceWithQts param not present" do
+        let(:params) { { qualifications: "Other,QtsOnly" } }
+        it { is_expected.to eq(false) }
+      end
+
+      context "when PgdePgceWithQts param is present" do
+        let(:params) { { qualifications: "QtsOnly,PgdePgceWithQts" } }
+        it { is_expected.to eq(true) }
+      end
+
+      context "when qualifications is empty" do
+        let(:params) { { qualifications: "" } }
+        it { is_expected.to eq(true) }
+      end
+
+      context "when qualifications is nil" do
+        let(:params) { {} }
+        it { is_expected.to eq(true) }
+      end
+    end
+
+    describe "pgde_pgce_with_qts_checked" do
+      subject { described_class.new(params: params).other_checked? }
+
+      context "when Other param not present" do
+        let(:params) { { qualifications: "QtsOnly,PgdePgceWithQts" } }
+        it { is_expected.to eq(false) }
+      end
+
+      context "when Other param is present" do
+        let(:params) { { qualifications: "QtsOnly,Other" } }
+        it { is_expected.to eq(true) }
+      end
+
+      context "when qualifications is empty" do
+        let(:params) { { qualifications: "" } }
+        it { is_expected.to eq(true) }
+      end
+
+      context "when qualifications is nil" do
+        let(:params) { {} }
```

Comment:
did you meant 

`        let(:params) { { qualifications: nil } }`

Reply:
No. I meant, when the key doesn't exist. I'll change the description of the context.

Reply:
Fixed